cmd/space-daemon/main.go

```diff
@@ -109,6 +113,33 @@ func main() {
 	}
 }
 
+func setULimit() {
+	if runtime.GOOS == "windows" {
+		// Only available on UNIX systems
+		return
+	}
+
+	var rLimit syscall.Rlimit
+
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		log.Error("Error Getting Rlimit. Please run `ulimit -n 1000` from a privileged user to avoid issues when running the space daemon.", err)
+		return
+	}
+	log.Debug(fmt.Sprintf("Got Rlimit: Cur: %d, Max: %d", rLimit.Cur, rLimit.Max))
+
+	// Max allowed value is 10240 even when rLimit.Max can go beyond that
+	rLimit.Cur = uint64(math.Min(10240, float64(rLimit.Max)))
+
+	err = syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
```

Comment:
Is this on a per-process basis? Just trying to think if this impacts users with machines with low mem or a leak happens. But we can probably deal with that if it happens.

Another thing was that the `lsof` dump from @giankotarola had textile having about 160 files open re: it's `4001` port. That's weird because @perfectmak put in a fix to limit textile to 100 by default so not sure how that could be over.  Just putting that out there but maybe not worth it now to look further and we can come back to this later.

Reply:
I'm a bit worried about different platforms.

This is how https://github.com/ipfs/go-ipfs handles `ulimit`:

https://github.com/ipfs/go-ipfs/tree/master/cmd/ipfs/util
https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/util/ulimit.go#L46
https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/util/ulimit_unix.go
https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/util/ulimit_windows.go
https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/util/ulimit_freebsd.go

Reply:
Noticed this when the windows build failed. Sent a fix in 1c0e374. Agree we can test with this new settings and go from there.